File: 1929-concatenation-of-array/1929-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n*2];
        //if n = 3, ans.length = 6
        //if n = 2, ans.length = 4
        
        for(int i = 0; i < n; i++){
            ans[i] = nums[i];
        }

        for(int i = 0; i < n; i++){
            ans[i+n] = nums[i];
        }
        return ans;
    }
}

File: 1929-concatenation-of-array/1929-concatenation-of-array.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * (2 * n) #populate 'ans' arr with 0's with twice the length of 'nums'
        
        for i in range(n):  #first half of the 'ans' arr is populated
            ans[i] = nums[i]
            
        for o in range(n):  #second half of array starts at length of original array 
            ans[n + o] = nums[o]
            
        return ans
    
    